### ISSUE TITLE
Qcubed over https on nginx

### ISSUE THREAD

**Issue by neuroticimbecile** [qcubed/framework#704]
Hi Qcubed,

I'm trying to make qcubed (just the sample.php file) run over https on nginx.
When I access the page over https, the sample.php script does not work.
However, it works flawlessly when accessed via plain http.

Are there any special configs needed to make qcubed work over https?

This is on QCubed 2.2.3

Thanks,
-eric

**Comment by matthiaz** [comment]
Eric,

I'm running a couple full webapplications with qcubed over https myself. No
problem there.
I haven't tested sample.php though. But then again, I never run https on my
local development pc.

Matthias

2015-06-09 11:42 GMT+02:00 neuroticimbecile <notifications@github.com>:

**Comment by neuroticimbecile** [comment]
Hi Mathias,

Thank you very much for your reply.

Let me provide some more details:
. the sample.php script loads fine when accessed over https, but nothing
happens when the "Click Me" button is clicked.
. it will only work when the page is accessed over plain http, the label
will change to "Hello, World!" when the button is clicked

What I've tried:
. previously, the qcubed installer was run over http, while the sample site
was accessed over https
. so I started over again and ran the qcubed installer over https
. the sample.php script still loads fine, but the button still would not
work
. I ran a diff against the old installation, and there are no differences
except for these files:
- .git/logs/HEAD
- .git/logs/refs/heads/master
- .git/objects/pack
- .git/packed-refs
. i also ran a codegen, and I noticed that the buttons in the generated
"drafts" scripts also don't work over https

BTW,
I am running nginx on a non-standard port, so the sample script is accessed
at e.g. https://server_address:9999/sample.php

Any ideas how to diagnose the cause of this problem?

Thanks,
-eric

**Comment by matthiaz** [comment]
Hi Eric,

If I were to have this kind of problem I would use firebug (see
http://getfirebug.com/) to look at the requests the button is doing. Is it
still going to the same address? Or did the port change?
Note: According to the github the sample.php hasn't been changed since
august 2014. So it might be it simply doesn't work anymore because of other
changes.
@vaibhav-kaushal <https://github.com/vaibhav-kaushal> made that last
change, maybe he knows what's going on.

Matthias


2015-06-10 9:26 GMT+02:00 neuroticimbecile <notifications@github.com>:

**Comment by neuroticimbecile** [comment]
Hi Matthias,

Thank you for your patience.
I was able to fix the problem by using local jQuery files
in configuration.inc.php:

=====
define ('__JQUERY_BASE__',
 'jquery/jquery.min.js,jquery/jquery-migrate.min.js');
define ('__JQUERY_EFFECTS__',   'jquery/jquery-ui.custom.min.js');
=====

Regards,
-eric

**Comment by matthiaz** [comment]
Ooooh,

Then I probably know what is going on.

The jquery files get loaded from a remote location that uses http.
define ('__JQUERY_BASE__', '
http://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js,jquery/jquery-migrate.min.js'
);   define ('__JQUERY_EFFECTS__', '
http://ajax.googleapis.com/ajax/libs/jqueryui/1.9.2/jquery-ui.min.js');

When you are navigating on https and are suddenly loading stuff from http.
Your webbrowser gives an error message.

So your work around is perfect.




2015-06-10 11:10 GMT+02:00 neuroticimbecile <notifications@github.com>:

**Comment by neuroticimbecile** [comment]
Thanks for the explanation, Matthias!
I'm now going to close this thread. I hope it will help someone in the future.
